modules/storages/app/services/storages/file_links/copy_file_links_service.rb

```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2024 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  module FileLinks
+    class CopyFileLinksService
+      def self.call(source:, target:, user:, work_packages_map:)
+        new(source:, target:, user:, work_packages_map:).call
+      end
+
+      def initialize(source:, target:, user:, work_packages_map:)
+        @source = source
+        @target = target
+        @user = user
+        @work_packages_map = work_packages_map.to_h { |key, value| [key.to_i, value.to_i] }
+      end
+
+      def call
+        source_file_links = FileLink
+          .includes(:creator)
+          .where(container_id: @work_packages_map.keys, container_type: "WorkPackage")
```

Comment:
There will be a call to `find_each` which will immediately return in case of an empty result.

In all cases it calls for a query or iterates over the empty array, so no extra handling should be necessary or distracting in this case.